app/src/main/java/com/example/androidacademyhw/movies/FragmentMoviesList.kt

```diff
@@ -22,47 +29,43 @@ class FragmentMoviesList : Fragment() {
     private val binding get() = _binding!!
 
     private lateinit var moviesAdapter: MoviesAdapter
-    private val actors = mutableListOf<Actor>()
-    private val movies = mutableListOf<Movie>()
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
-        val view = inflater.inflate(R.layout.fragment_movies_list, container, false)
         _binding = FragmentMoviesListBinding.inflate(inflater, container, false)
         return binding.root
-
-        return view
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
-        initData()
-        initAdapter()
+        GlobalScope.launch(Dispatchers.Main) {
+            val movies = getMovies(requireContext())
+            initAdapter(movies)
+        }
     }
 
     override fun onDestroy() {
         super.onDestroy()
         _binding = null
     }
 
-    private fun initAdapter() {
+    private fun initAdapter(movies: List<Movie>) {
         moviesAdapter =
             MoviesAdapter(
                 object : MoviesAdapter.OnMovieClickListener {
                     override fun onMovieClick(movie: Movie) {
                         onMovieDetailsClicked(movie)
                     }
-
                 }
             )
 
-        moviesAdapter.likeClickListener = { position, isLike ->
-            onLikeClick(position, isLike)
-        }
+//        moviesAdapter.likeClickListener = { position, isLike ->
+//            onLikeClick(position, isLike)
+//        }
```

Comment:
remove if do not need it anymore, it is a bad practice to have commented code

Reply:
Done